code/modules/mob/living/simple_animal/revenant/revenant_abilities.dm

```diff
@@ -0,0 +1,318 @@
+/obj/effect/proc_holder/spell/targeted/revenant_life_tap
+	name = "Unlock: Life Tap (25E)"
+	desc = "Draws from your own life tool to gain more essence. Can only be cast three times."
+	panel = "Revenant Abilities (Locked)"
+	charge_max = 600
+	clothes_req = 0
+	range = -1
+	include_user = 1
+	var/locked = 1
+
+/obj/effect/proc_holder/spell/targeted/revenant_life_tap/cast(list/targets, var/mob/living/simple_animal/revenant/user = usr)
+	if(user.inhibited)
+		user << "<span class='warning'>Something is blocking the use of [src]!</span>"
+		charge_counter = charge_max
+		return
+	if(locked && essence_check(25, 1))
+		user << "<span class='info'>You have unlocked Life Tap!</span>"
+		name = "Life Tap (0E)"
+		panel = "Revenant Abilities"
+		locked = 0
+		charge_counter = charge_max
+		return
+	if(locked)
+		charge_counter = charge_max
+		return
+	for(var/mob/living/simple_animal/revenant/target in targets)
+		if(!target.strikes)
+			target << "<span class='warning'>Your life force has grown too weak to life tap again.</span>"
+		target.strikes--
+		target << "<span class='info'>You convert your own life into energy.[target.strikes ? "" : " This is the last time you can do this."]</span>"
+		target.change_essence_amount(50, 0, 0, "your life pool")
+
+
+//Seed of Draining: Plants a 'seed' in the target that will slowly siphon essence silently from them.
+/obj/effect/proc_holder/spell/targeted/revenant_seed_drain
+	name = "Unlock: Seed of Draining (20E)"
+	desc = "Corrupts a target with dark energies. Their essence will slowly drain for some time."
+	panel = "Revenant Abilities (Locked)"
+	charge_max = 1200
+	clothes_req = 0
+	range = 1 //Adjacent
+	include_user = 1
+	var/locked = 1
+	var/planted = 0
+
+/obj/effect/proc_holder/spell/targeted/revenant_seed_drain/cast(list/targets, var/mob/living/simple_animal/revenant/user = usr)
+	if(user.inhibited)
+		user << "<span class='warning'>Something is blocking the use of [src]!</span>"
+		charge_counter = charge_max
+		return
+	if(locked && essence_check(20, 1))
+		user << "<span class='info'>You have unlocked Seed of Draining!</span>"
+		charge_counter = charge_max
+		name = "Seed of Draining (20E)"
+		locked = 0
+		panel = "Revenant Abilities"
+		include_user = 0
+		return
+	if(locked)
+		charge_counter = charge_max
+		return
+	if(!essence_check(20))
+		charge_counter = charge_max
+		return
+	if(planted)
+		user << "<span class='warning'>You are already passively draining essence.</span>"
+		charge_counter = charge_max
+		return
+	for(var/mob/living/carbon/human/target in targets)
+		if(target.stat == DEAD)
+			user << "<span class='warning'>[target] is dead and will not yield essence.</span>"
+			charge_counter = charge_max
+		user << "<span class='info'>You plant a draining seed on [target].</span>"
+		planted = 1
+		for(var/i = 0, i < 120, i++)
+			sleep(10)
+			user.essence++ //Drains 120 essence total
+			if(prob(1))
+				target << "You feel a curious draining sensation."
```

Comment:
120 essence is quite a lot.
this should probably induce some staminaloss over time instead of a dumb 1% chance of the equivalent of "you feel a tiny prick!"